### DISCUSSION TITLE
How to Status Check Servers in Docker Containers?

### DISCUSSION THREAD

**Discussion by JohnHind** [Lissy93/dashy#1580] [Help and Support] [answered]
I have Dashy and a number of other services running in Docker Containers on a Raspberry Pi and also a Python FastAPI server which runs outside of Docker on the same Pi. I am accessing the Dashy dashboard from a Vivaldi browser on a Windows machine.

After a lot of head-banging I was able to get Dashy Status Checks running for the FastAPI server, but getting it running for the services in the other Docker Containers still eludes me. I was wondering if anyone could give me any tips.

For the FastAPI server, the key was to use a Status Check URL of the form: 'http://172.17.0.1:8000/' when the Service URL is of the form 'http://octosnap.local:8000/'. After re-reading the documentation about six times I eventually understood this difference is because the Service URL is used by the browser on the Windows machine while the Status Check URL is used by the Dashy server in its Docker Container on the Pi. From a Google search, the Docker Virtual Network uses the standard IP 172.17.0.1 for the physical machine on which Docker is running.

But by the same logic, to refer to one Docker container from within another, the container name ought to work. The following Service URL responds fine for Portainer in a Docker Container: http://octosnap.local:9000/, but the corresponding Status Check URL: http://portainer:9000/ is not responsive (the dot remains grey). I checked the name 'portainer' in the Portainer containers list and it is running with that name. I also tried the numeric IP address given for that container, but it still gives Request timed out.

**Comment by lissy93** [comment]
Have you seen the docs in [Troubleshooting --> Status Checks Failing](https://github.com/Lissy93/dashy/blob/master/docs/troubleshooting.md#status-checks-failing)? I'm not sure there is much beyond that, that I can add here, without seeing what errors you're getting in the [browser console](https://github.com/Lissy93/dashy/blob/master/docs/troubleshooting.md#how-to-open-browser-console). The answer has probably been covered in a [previous issue](https://github.com/Lissy93/dashy/issues?q=is%3Aissue+status+check), so that might be worth looking at too.

**Comment by JohnHind** [comment, marked as answer]
Thanks Alicia (and also for Dashy itself!). This hot tip: https://github.com/Lissy93/dashy/discussions/445 eventually got me there and I've added a fuller information to that tip. Unfortunately it did not have the magic phrase 'Status Checks' in the heading, or I'd have found it earlier!
